utils/crypto/bls/signer.go

```diff
@@ -0,0 +1,10 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package bls
+
+type Signer interface {
+	PublicKey() *PublicKey
+	Sign(msg []byte) *Signature
+	SignProofOfPossession(msg []byte) *Signature
```

Comment:
I feel like this interface looks cleaner if we just merge these two function signatures and just add the ciphertext as a parameter. Having a separate function signature for each ciphertext feels like it just bloats the interface... but not sure what others think.

Reply:
I actually think that we should have a `ProofOfPossession()` method. We shouldn't really be doing proof-of-possession signatures of anything other than the pubkey. 

Regardless, that change would be beyond the scope of this PR.

Reply:
Just to elaborate here, the reason that I prefer to have both APIs over a single function where one has to pass in the digest is because I think this should be opaque to the caller. It's an implementation detail that these two methods only differ in the digest used for signing.